planner/optimize.go

```diff
@@ -75,6 +102,111 @@ func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is in
 	return plannercore.DoOptimize(ctx, builder.GetOptFlag(), logic)
 }
 
+func addHint(ctx sessionctx.Context, stmtNode ast.StmtNode) (ast.StmtNode, *bindinfo.HintsSet) {
+	// When the domain is initializing, the bind will be nil.
+	if ctx.Value(bindinfo.SessionBindInfoKeyType) == nil {
+		return stmtNode, nil
+	}
+	switch x := stmtNode.(type) {
+	case *ast.ExplainStmt:
+		var oriHintSet *bindinfo.HintsSet
+		switch x.Stmt.(type) {
+		case *ast.SelectStmt:
+			normalizeExplainSQL := parser.Normalize(x.Text())
+			idx := strings.Index(normalizeExplainSQL, "select")
```

Comment:
`explain` statement supports not only `select` statements.

Reply:
But sql binding only supports select now.

Reply:
This is a little tricky, can we use the restored SQL from the AST?

Reply:
No, the `Restore` only guranteens that the meaning is the same, not the text.